system/ee/legacy/core/Input.php

```diff
@@ -374,55 +376,56 @@ function server($index = '', $xss_clean = FALSE)
 	*/
 	function ip_address()
 	{
-		if ($this->ip_address !== FALSE)
-		{
+		if ($this->ip_address !== false) {
 			return $this->ip_address;
 		}
 
-		if (REQ == 'CLI')
-		{
-			return '0.0.0.0';
+		if (REQ === 'CLI') {
+			return self::DEFAULT_IP_ADDRESS;
 		}
 
+		$possibleHeaders = [
+			'HTTP_CF_CONNECTING_IP', // Cloudflare IP forwarding
+			'HTTP_X_FORWARDED',
+			'HTTP_X_FORWARDED_FOR',
+			'HTTP_FORWARDED',
+			'HTTP_FORWARDED_FOR',
+			'HTTP_CLIENT_IP',
+			'HTTP_X_CLIENT_IP',
+			'HTTP_X_CLUSTER_CLIENT_IP',
+		];
+
 		$proxy_ips = config_item('proxy_ips');
-		if ( ! empty($proxy_ips))
-		{
+		$this->ip_address = $_SERVER['REMOTE_ADDR'];
```

Comment:
Changed the logic by moving this up and setting the default. There was an unnecessary `else` in the conditional below.